packages/legacy/core/App/components/record/Record.tsx

```diff
@@ -38,7 +39,12 @@ const Record: React.FC<RecordProps> = ({ header, footer, fields, hideFieldValues
   })
 
   const resetShown = (): void => {
-    setShown(fields.map(() => false))
+    setShown(fields.map(() => showAll))
+    setShowAll(!showAll)
+  }
+
+  const toggleShownState = (showState: boolean[]): void => {
```

Comment:
Are you able to give `x` a more descriptive variable name and scope the if logic. It's a little more clear and follows the convention used throughout the app.
`
```
if (...) { 
   setShowAll(!showAll)
}
````